tket/test/src/Circuit/test_Circ.cpp

```diff
@@ -1326,6 +1326,41 @@ SCENARIO("Test depth_by_type method") {
     REQUIRE(circ.depth_by_types({OpType::CX, OpType::CY}) == 2);
   }
 }
+SCENARIO("Test depth_by_2q_gates method") {
+  GIVEN("2q OpTypes") {
+    Circuit circ(3, 1);
+    circ.add_op<unsigned>(OpType::CX, {0, 1});
+    circ.add_op<unsigned>(OpType::Z, {0});
+    circ.add_op<unsigned>(OpType::Z, {1});
+    circ.add_op<unsigned>(OpType::CZ, {1, 0});
+    circ.add_conditional_gate<unsigned>(OpType::CY, {}, {1, 2}, {0}, 0);
+    REQUIRE(circ.depth_by_2q_gates() == 2);
```

Comment:
Maybe I understand this check wrong, but I would expect a depth of 3 because CX, CZ, CY all act on qubit 1?